### PR TITLE
fau_local 2022feb07 km

### PR DESCRIPTION
**Pull request by miccichek**
Updated to include FAU_STG_EXT.1/LocSpace SFRs, AAs, and other changes.

### CODE REVIEW

File: NDcPP_v2_2e.adoc [hunk 1]
```diff
@@ -907,13 +908,19 @@ _For pNDs, ‘on the TOE itself’ or ‘locally’ means on storage inside or d
...
 _For vNDs, local storage is any storage accessible by TOE software. In a virtualized environment, ‘local’ storage is under the control of the VS and may be physically located on the local host, but it could also be located on a network drive or storage array._
 
-*FAU_STG_EXT.1.3* The TSF shall [selection: _drop new audit data, overwrite previous audit records according to the following rule: [assignment: rule for overwriting previous audit records], [assignment: other action]_] when the local storage space for audit data is full.
+===== FAU_ STG_EXT.1/LocSpace Local Audit Event Storage
+
+*FAU_STG_EXT.1/LocSpace Local Audit Event Storage*
+
+*FAU_STG_EXT.1.1/LocSpace* The TSF shall maintain a [_selection: log file, database, buffer, assignment:[other local logging method]_] of audit records in the event that an interruption of communication with the remote syslog server occurs.
+
+*FAU_STG_EXT.1.2/LocSpace* The TSF shall be able to store [_selection: persistent, non-persistent_] audit records locally with a minimum storage size of [_assignment: number of records, file/buffer size(s)_].
```
Comment (plughy2) [inline NDcPP_v2_2e.adoc:917]: Would it be better if the SFR and associated Assurance Activities used the terms " non-volatile" and "volatile" storage?  

For example:  The TSF shall be able to store audit records locally in [_selection: non-volatile, volatile_] storage with a minimum storage size of [_assignment: number of records, file/buffer size(s)_]

File: ND Supporting Document 2_2.adoc [hunk 2]
```diff
@@ -322,57 +322,87 @@ In general, when using open source libraries like OpenSSL, passing on error mess
...
+. If more than one selection is made for FAU_STG_EXT.1.3/LocSpace, the evaluator shall review the Guidance Documentation to ensure that it contains instructions on specifying which action is performed when the local storage space is full
+
+
+===== Tests
+
+[arabic, start=49]
+. The evaluator shall perform the following tests:
+[loweralpha, start=1]
+.. Test 1: The evaluator shall access the local audit trail without authentication as Security Administrator (either by authentication as a non-administrative user, if supported, or without authentication at all) and attempt to modify and delete the audit records. The evaluator shall verify that these attempts fail. If the interface cannot be accessed to attempt the command due to permissions or configuration, this test requirement is satisfied.
+.. (Conditional – if local log can be viewed) Test 2: The evaluator shall perform operations that generate audit data and verify that this data is stored locally. The evaluator shall then make note of whether the TSS claims persistent or non-persistent logging and perform one of the following actions:
```
[the remark this answers is not in the document]
Comment (miccichek) [reply ND Supporting Document 2_2.adoc:370]: Removed the conditional statement. While some products may not have this feature if they are an endpoint node in a distributed TOE configuration, a justification could be made that the local viewing would then be through the main management device.

File: ND Supporting Document 2_2.adoc [hunk 3]
```diff
@@ -322,57 +322,87 @@ In general, when using open source libraries like OpenSSL, passing on error mess
...
+. If the storage size is configurable, the evaluator shall review the Guidance Documentation to ensure that it contains instructions on specifying the required parameters.
+. If more than one selection is made for FAU_STG_EXT.1.3/LocSpace, the evaluator shall review the Guidance Documentation to ensure that it contains instructions on specifying which action is performed when the local storage space is full
+
+
+===== Tests
+
+[arabic, start=49]
+. The evaluator shall perform the following tests:
+[loweralpha, start=1]
+.. Test 1: The evaluator shall access the local audit trail without authentication as Security Administrator (either by authentication as a non-administrative user, if supported, or without authentication at all) and attempt to modify and delete the audit records. The evaluator shall verify that these attempts fail. If the interface cannot be accessed to attempt the command due to permissions or configuration, this test requirement is satisfied.
```
[the remark this answers is not in the document]
Comment (miccichek) [reply ND Supporting Document 2_2.adoc:369]: The tests map to FAU_STG_EXT.1/LocSpace as a whole. Added Test 5 (to discuss) relating to FAU_STG_EXT.1 on interrupting the connection and verifying logs are captured.

File: ND Supporting Document 2_2.adoc [hunk 4]
```diff
@@ -322,57 +322,87 @@ In general, when using open source libraries like OpenSSL, passing on error mess
...
+.. (Conditional – if local log can be viewed) Test 2: The evaluator shall perform operations that generate audit data and verify that this data is stored locally. The evaluator shall then make note of whether the TSS claims persistent or non-persistent logging and perform one of the following actions:
 [arabic]
-... The audit data remains unchanged with every new auditable event that should be tracked but that the audit data is recorded again after the local storage for audit data is cleared (for the option ‘drop new audit data’ in FAU_STG_EXT.1.3).
-... The existing audit data is overwritten with every new auditable event that should be tracked according to the specified rule (for the option ‘overwrite previous audit records’ in FAU_STG_EXT.1.3)
-... The TOE behaves as specified (for the option ‘other action’ in FAU_STG_EXT.1.3).
+... If persistent logging is claimed, the evaluator shall perform a power cycle of the TOE and ensure that following power on operations the log events generated are still maintained within the local audit storage.
```
[the remark this answers is not in the document]
Comment (miccichek) [reply ND Supporting Document 2_2.adoc:372]: Updated

File: ND Supporting Document 2_2.adoc [hunk 5]
```diff
@@ -322,57 +322,87 @@ In general, when using open source libraries like OpenSSL, passing on error mess
...
+... If non-persistent logging is claimed, the evaluator shall perform a power cycle of the TOE and ensure that following power on operations the log events generated are no longer present within the local audit storage unless otherwise claimed in the TSS.
```
[the remark this answers is not in the document]
Comment (miccichek) [reply ND Supporting Document 2_2.adoc:373]: Updated item 1, removed item 2

File: NDcPP_v2_2e.adoc (hunk 1 above)
Comment (miccichek) [reply NDcPP_v2_2e.adoc:917]: I do not think we should change this. I believe the use of persistent vs. volatile is appropriate when it comes to discussing logging.

**Review by plughy2 — APPROVED** [review]

**Review by kr15tyk — APPROVED** [review]

**Pull request merged**